docs/standard/io/buffers.md

```diff
@@ -0,0 +1,188 @@
+|`ReadOnlySequenceSegment<T>`|`Length`|`O(1)`
+|`ReadOnlySequenceSegment<T>`|`GetPosition(long)`|`O(number of segments)`
+|`ReadOnlySequenceSegment<T>`|`Slice(int, int)`|`O(number of segments)`
+|`ReadOnlySequenceSegment<T>`|`Slice(SequencePostion, SequencePostion)`|`O(1)`
+
+Because of this mixed representation, the `ReadOnlySequence<T>` exposes indexes as `SequencePosition` instead of an integer. A `SequencePosition`:
+
+- Is an opaque value that represents an index into the `ReadOnlySequence<T>` where it originated.
+- Consists of two parts, an integer and an object. What these two values represent are tied to the implementation of `ReadOnlySequence<T>`.
+
+### Access data
+
+The `ReadOnlySequence<T>` exposes data as an enumerable of `ReadOnlyMemory<T>`. Enumerating each of the segments can be done using a basic foreach:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet3)]
+
+The preceding method searches each segment for a specific byte. If you need to keep track of each segment's `SequencePosition`, 
+<xref:System.Buffers.ReadOnlySequence%601.TryGet%2A?displayProperty=nameWithType> is more appropriate. The next sample changes the preceding code to return a `SequencePosition` instead of an integer. Returning a `SequencePosition` has the benefit of allowing the caller to avoid a second scan to get the data at a specific index.
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet4)]
+
+The combination of `SequencePosition` and `TryGet` acts like an enumerator. The position field is modified at the start of each iteration to be start of each segment within the `ReadOnlySequence<T>`.
+
+The preceding method exists as an extension method on `ReadOnlySequence<T>`. <xref:System.Buffers.BuffersExtensions.PositionOf%2A> can be used to simplify the preceding code:
+
+```csharp
+SequencePosition? FindIndexOf(in ReadOnlySequence<byte> buffer, byte data) => buffer.PositionOf(data);
+```
+
+#### Process a ReadOnlySequence\<T\>
+
+Processing a `ReadOnlySequence<T>` can be challenging since data may be split across multiple segments within the sequence. For the best performance, split code into two paths:
+
+- A fast path that deals with the single segment case.
+- A slow path that deals with the data split across segments.
+
+There are a few approaches that can be used to process data in multi-segmented sequences:
+
+- Use the [`SequenceReader<T>`](#sequencereadert).
+- Parse data segment by segment, keeping track of the `SequencePosition` and index within the segment parsed. This avoids unnecessary allocations but may be inefficient, especially for small buffers.
+- Copy the `ReadOnlySequence<T>` to a contiguous array and treat it like a single buffer:
+  - If the `ReadOnlySequence<T>` has a length less than 256, it may be reasonable to copy the data into a stack-allocated buffer using the [stackalloc](../../csharp/language-reference/operators/stackalloc.md) operator.
```

Comment:
I wouldn't say a specific number here for the length, T could be a large object and stackalloc may not be a good idea. 

```suggestion
  - If the size of the `ReadOnlySequence<T>` is small, it may be reasonable to copy the data into a stack-allocated buffer using the [stackalloc](../../csharp/language-reference/operators/stackalloc.md) operator.
```